mlrun/runtimes/nuclio/function.py

```diff
@@ -958,6 +957,38 @@ def invoke(
             data = json.loads(data)
         return data
 
+    def _with_sidecar(self, name: str, image: str, port: int = None):
+        """
+        Add a sidecar container to the function pod
+        :param name:    Sidecar container name
+        :param image:   Sidecar container image
+        :param port:    Sidecar container port
+        """
+        # TODO: validate image on server side
+        name = name or f"{self.metadata.name}-sidecar"
+        sidecar = self._set_sidecar(name)
+        if image:
+            sidecar["image"] = image
+
+        if port:
+            sidecar["ports"] = [
```

Comment:
It looks like the sidecar function in nuclio supports multiple ports. Why not expose it here, and allow sending a list of ports to this function? Maybe change `port` to be `ports: Optional[Union[int, list[int]]] = None`?